### DISCUSSION TITLE
Left join alias missing on delete query

### DISCUSSION THREAD

**Discussion by pgcm** [doctrine/doctrine2#10935] [Support Questions] [answered]
Hello all,

I'm trying to delete a record, and as a failsafe, I want to make sure at deletion time other tables do not have any reference. So my initial idea was to `left join` and verify the join result is null.

```php
    public function delete(int $myId): mixed
    {
        return $this->createQueryBuilder('firstTable')
            ->delete(FirstTable::class, 'firstTable')
            ->leftJoin(SecondTable::class, 'secondTable', Join::WITH, 'firstTable.id = secondTable.myId')
            ->where('firstTable.id = :myId')
            ->andWhere('secondTable.id is null')
            ->setParameter('myId', $myId, ParameterType::INTEGER)
            ->getQuery()
            ->getResult();
    }
```

The problem is that the alias of the join does not seem to be taken into consideration, I'm having the following error:
` Error: 'secondTable' is not defined.`

Is this the expected behavior?
Thank you.

**Comment by derrabus** [comment, marked as answer]
> Is this the expected behavior?

Yes. There is no JOIN in DELETE queries.

Furthermore you're reinventing foreign key constraints. You should not need this "failsafe" code of yours.

**Reply by pgcm** [reply]
Thanks for the answer. 

> Furthermore you're reinventing foreign key constraints. You should not need this "failsafe" code of yours.

Totally agree 😞
